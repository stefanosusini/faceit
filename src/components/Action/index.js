import React, { memo } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import styles from './index.module.scss';

let Action = ({ className, label, disabled, onClick }) => {
  const handleClick = (evt) => {
    evt.preventDefault();
    if (onClick && !disabled) onClick(evt);
  };

  return (
    <button className={classNames(styles.container, className)}
            onClick={handleClick}
            disabled={disabled}>
      {label}
    </button>
  );
};

Action = memo(Action);

Action.propTypes = {
  className: PropTypes.string,
  label: PropTypes.node,
  disabled: PropTypes.bool,
  onClick: PropTypes.func.isRequired,
};

export default Action;
