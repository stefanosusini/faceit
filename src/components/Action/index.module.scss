@import "scss/mixins";

.container {
  display: flex;
  position: relative;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  flex-grow: 0;
  padding: 0 0.75em;
  box-shadow: inset 0 0 1px 0 white;
  border: none;
  background: transparent;
  outline: none;
  font-size: 1em;
  @include responsiveRowHeight();

  &:not([disabled]) {
    &:hover {
      background-color: #444;
    }
    &:active {
      background-color: #333;
    }
  }
}
