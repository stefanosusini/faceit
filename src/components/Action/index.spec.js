import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Action from './index';

describe('Action component', () => {
  it('should render', () => {
    const wrapper = shallow(<Action label="Action" onClick={() => {}} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  describe('if className is provided', () => {
    it('should render with the class name', () => {
      const wrapper = shallow(<Action label="Action" className="CLASSNAME" onClick={() => {}} />);
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
  describe('when clicked', () => {
    it('should call the onClick handler', () => {
      const handler = jest.fn();
      const wrapper = shallow(<Action label="Action" onClick={handler} />);
      wrapper.find('button').simulate('click', { preventDefault: () => {} });
      expect(handler).toHaveBeenCalled();
    });
    describe('if disabled and clicked', () => {
      it('should NOT call the onClick handler', () => {
        const handler = jest.fn();
        const wrapper = shallow(<Action label="Action" onClick={handler} disabled />);
        wrapper.find('button').simulate('click', { preventDefault: () => {} });
        expect(handler).not.toHaveBeenCalled();
      });
    });
  });
});
