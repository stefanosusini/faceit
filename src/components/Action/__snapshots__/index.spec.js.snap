// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Action component if className is provided should render with the class name 1`] = `
<button
  className="container CLASSNAME"
  onClick={[Function]}
>
  Action
</button>
`;

exports[`Action component should render 1`] = `
<button
  className="container"
  onClick={[Function]}
>
  Action
</button>
`;
